public/store/reducers/workspace_reducer.ts

```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { createSlice } from '@reduxjs/toolkit';
+
+const initialState = {
+  isDirty: false,
+  components: [],
+};
+
+const workspaceSlice = createSlice({
```

Comment:
Silly questions: But don't we have to treat this as an actionType?

Reply:
`createSlice` takes care of that for us, where we don't need to create action strings at all. See https://redux.js.org/tutorials/fundamentals/part-8-modern-redux#writing-slices

Reply:
Oh! Something new for me. I'll read about it.